### PR TITLE
Add logger

### PR DESCRIPTION
**Pull request by mattijsbliek**
This PR adds the [Signale](https://github.com/klaussinani/signale) via a simple interface, so it can easily be swapped out for something else.

The log level is configurable via `usersettings.json`. To make these settings readily available throughout the application, I wrote a small config utility class which loads the settings on startup.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,8 +2,9 @@
   "name": "sonos-nfc",
   "version": "0.1.0",
   "description": "Use NFC tags to control Sonos - a node app inspired by and compatible with hankhank10/vinylemulator, with better NFC reader compatibility",
-  "main": "sonos_nfc.js",
+  "main": "index.js",
```
Comment (mattijsbliek) [inline package.json:5]: I saw I forgot to change this when moving `sonos_nfc` to the `lib` folder.

**Comment by mattijsbliek** [comment]
Noticed just now that the tests won't run due to the `import.meta.url` in the config class. Will see if I can find a fix somewhere this week.

**Comment by ryanolf** [comment]
Great. I’ll look at setting up some CI stuff on the repo to run tests
automatically on pull requests to make catching this sort of thing easier.

**Comment by mattijsbliek** [comment]
Nice! Tests are fixed 👍

**Comment by ryanolf** [comment]
I'm getting failed tests for `process_sonos_command` on this. The error is from line 100 of process_sonos_command.js and is:
```
TypeError: _logger.default.success is not a function
```

I just setup CI to run the test suite on the main branch and pull requests against it. I'm going to close and open this pull request to see if I can get it to run.

**Comment by ryanolf** [comment]
It looks like that `logger.success` is supposed to be a `logger.info`.

**Comment by ryanolf** [comment]
@mattijsbliek Tests pass by changing that `logger.success` to `logger.info` but the code doesn't run for me. The first issue seems to be that logger.js is looking for a module `config` but it should be `config.js`. With that fixed, the "Unsupported log_level" exception is thrown -- I guess it's not reading the config quite right. Is this running for you as-is?

I have a an additional thought. Should we have `usersettings.json` be part of the repo? If folks have their own settings there, they probably don't want them overwritten. Here's a suggestion: we move `usersettings.json` to `usersettings.dist.json` and in your settings class, let's load both and have the values in `usersettings.json` override those in `usersettings.dist.json`. If folks want to customize, we ask them to copy `usersettings.dist.json` to `usersettings.json` and make changes there (maybe just to the variables they want to change). This allows us to add new settings and defaults going forward without messing with customizations. Thoughts?

If this seems reasonable, in addition to making sure the code runs and tests pass, could you make this change?

**Comment by mattijsbliek** [comment]
Sorry about the test and the runtime, sloppy on my part. Both are fixed now :)

Like the idea of making it easy for folks to add their own user settings file 👍 I'm wondering if it would make more sense to go for an `.env` file instead? This would allow consumers more flexibility in whether to set them through a `.env` or other via other means. We could provide a `.env.example` that people can copy. I would probably avoid reading from that file though, and would just define config defaults in the config class itself.

This could also work with a `usersettings.json` btw, maybe an example file isn't even needed since the values are in the documentation. If we move the default values into the config file, we can get rid of `usersettings.json` completely, and just tell folks to create one and define overrides if needed.

No strong opinion about either of these solutions btw, so happy to implement what you think is best.

**Comment by ryanolf** [comment]
@mattijsbliek Thanks for the fixes. The code runs and works now so I could pull this in. I think we should resolve the `usersettings.json` issue first to minimize the hassle in upgrading. Right now, if someone has customized `usersettings.json` they won't be able to pull this update (or any updates) trivially and will have to do a merge, which some folks may not know how to do. If people happen to keep their old version of `usersettings.json` instead of doing a merge (e.g. git stash/pop), the program would crash with this update. So let's think this through and make sure we solve the updates issue.

I made a mistake in my initial release by version controlling a customizable file. I think the ideal path forward rectifies this with as little work by any current users as possible. To this end, ideally I'd remove `usersettings.json` from the repo but allow folks who modified it to to do a simple fast-forward merge, keeping their own custom version. It appears this isn't possible.

So here's what I propose. We remove `usersettings.json` from the repo and add it to `.gitignore`. If folks have unmodified `usersettings.json` (unlikely-- they probably have customized the room at least I'd think) the file will be deleted. If it's modified, they will get a message that they have an unclean working directory on any `pull` and be advised to `stash` by git. In case people do that, I'd like to make sure that works, which means we should continue to honor the settings in `usersettings.json` at least for now. Maybe we "deprecate" it and add a log message encouraging people to change to using the new configuration.

I like the suggestion of having settings like the ones we have so far in a `.env` file, and I like the idea of providing a sample with comments. We can update the readme to show an example of how to set the room variable (which most people will want to do) via either `.env` or a command line variable.

As far as where to put defaults, I lean toward having defaults in a JSON separate from the config class. This would make loading `usersettings.json` pretty trivial. First load the default JSON. Then `Object.assign` the `usersettings.json` to override defaults. Then overwrite with relevant values from `process.env`. I don't feel strongly here, though. Having defaults in `config.js` could also be OK.

It seems like `dotenv` is the canonical package for loading `.env` files. Since we use `import` instead of `require` it seems like we need to either do something like [this](https://github.com/motdotla/dotenv/issues/133#issuecomment-255298822) or import `dotenv/config` directly. I think having the separate file (or maybe `config.js`?) load the module kinda makes sense.

Is this something you can work on and add to this PR?

**Comment by ryanolf** [comment]
@mattijsbliek If you [give me write access to this branch](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork) I may be able to help draft the readme or something.

**Pull request closed without merging**